fondant/component_spec.py

```diff
@@ -276,13 +283,17 @@ def to_file(self, path: t.Union[str, Path]) -> None:
                 sort_keys=False,
             )
 
+    def to_string(self) -> str:
+        """Return the component specification as a string."""
+        return json.dumps(self._specification)
+
     @property
     def input_arguments(self) -> t.Mapping[str, Argument]:
         """The input arguments of the component as an immutable mapping."""
         return types.MappingProxyType(
             {
                 info["name"]: Argument(
-                    name=info["name"],
+                    name=info["name"].replace("_", "-"),
```

Comment:
small nit but I noticed that the arguments were passed with a mix of underscore and upperscore (e.g. --input_manifest_path) so I changed this. However this might not be the best place to do this. I'll move the `replace` method to the `_add_and_parse_args` function

Reply:
I'm not sure if we even need to do this at all. This might be surprising for the users, instead of just using the name that they provided.

Reply:
The users still define argument with underscores in both the component function definition and specification. This only applies when running the component from the command line similar to [our boilerplate example](https://bitbucket.org/ml6team/nimbus/src/c798c261812e90c81e9aabab3b33ca603c09476f/src/_nimbus/boilerplates/mlpipelines/gcp/boilerplate/example_pipeline.py)

Not sure if there is a convention around this but I'm used to command line arguments with a mixture of hyphens and underscores always trip me up

Reply:
Agree that hyphens is nicer, but I think matching the arguments in the specification file is more important.